### DISCUSSION TITLE
Nuxt/Nitro not finding new files under /public

### DISCUSSION THREAD

**Discussion by TokyoPiglet** [nuxt/nuxt#25151] [Questions] [answered]
While starting my app with "npm run dev", I was able to create new files under the /public directory, and the app would be able to fetch the new file without re-starting the application. 

Now, I have deployed my app with "npx nuxt build" and previewed it with "node .output/server/index.mjs. I then added a new file under .output/public/new_data_file.json. When the app tries to $fetch the file, it emits this warning:
[Vue Router warn]: No match found for location with path "/new_data_file.json" The app is able to $fetch files that existed in the /public directory before I ran "npx nuxt build", but not the files I added after the build. 

Can someone kindly explain why this doesn't work? 

What is the right strategy to make new data files on the server's local storage available to the app?

**Comment by TokyoPiglet** [comment, marked as answer]
This seems to be a known limitation in Nitro. I found reading the local file using fs.createStream, and returning it as a stream works. 

I added an alias, "public", in nuxt.config.ts:

  alias: {
    "public": "/<rootDir>/public"
  },

and wrote a new API module, <rootDir>/server/api/read/[file].ts, as:

import fs from 'node:fs';
import { sendStream } from 'h3'
import { defineNuxtModule, resolvePath } from '@nuxt/kit'
import { join } from 'pathe'

export default defineEventHandler(async (event) => {
  const file = event.context.params.file;
  const entry = await resolvePath('public')
  return sendStream(event, fs.createReadStream(join(entry, file)))
});

**Comment by wyl20000205** [comment]
//  server/routes/images/[name].js    
//  http://localhost:3000/images/logo.png

import fs from "fs";
import path from 'path'
export default defineEventHandler(async (event) => {
    let dest = ''
    if (process.env.NODE_ENV == 'production') dest = '.output/public/images';
    let filePath = path.join(dest, event.context.params.name);
    return sendStream(event, fs.createReadStream(filePath));
});
